packages/clay-core/src/picker/Option.tsx

```diff
@@ -115,34 +121,66 @@ export function Option({
 
 	return (
 		<li role="presentation">
-			<button
-				{...otherProps}
-				{...hoverProps}
-				aria-describedby={ariaDescribedby}
-				aria-label={ariaLabel}
-				aria-labelledby={ariaLabelledby}
-				aria-posinset={ariaPosInset}
-				aria-selected={selectedKey === keyValue}
-				aria-setsize={ariaSetSize}
-				className={classNames('dropdown-item', {
-					active: selectedKey === keyValue,
-					focus: activeDescendant === keyValue && isFocus,
-					hover: activeDescendant === keyValue && !isFocus,
-				})}
-				disabled={disabled}
-				id={String(keyValue)}
-				onClick={() => onSelectionChange(keyValue!)}
-				role="option"
-				tabIndex={-1}
-			>
-				{selectedKey === keyValue && (
-					<span className="dropdown-item-indicator-start">
-						<Icon symbol="check-small" />
-					</span>
-				)}
-
-				{children}
-			</button>
+			{href ? (
+				<a
+					{...otherProps}
+					{...hoverProps}
+					aria-describedby={ariaDescribedby}
+					aria-label={ariaLabel}
+					aria-labelledby={ariaLabelledby}
+					aria-posinset={ariaPosInset}
+					aria-selected={selectedKey === keyValue}
+					aria-setsize={ariaSetSize}
+					className={classNames('dropdown-item', {
+						active: selectedKey === keyValue,
+						disabled,
+						focus: activeDescendant === keyValue && isFocus,
+						hover: activeDescendant === keyValue && !isFocus,
+					})}
+					href={href.toString()}
+					id={String(keyValue)}
+					onClick={() => onSelectionChange(keyValue!)}
+					role="option"
+					tabIndex={-1}
+				>
+					{selectedKey === keyValue && (
+						<span className="dropdown-item-indicator-start">
+							<Icon symbol="check-small" />
+						</span>
+					)}
+
+					{children}
+				</a>
+			) : (
+				<button
+					{...otherProps}
+					{...hoverProps}
+					aria-describedby={ariaDescribedby}
+					aria-label={ariaLabel}
+					aria-labelledby={ariaLabelledby}
+					aria-posinset={ariaPosInset}
+					aria-selected={selectedKey === keyValue}
+					aria-setsize={ariaSetSize}
+					className={classNames('dropdown-item', {
+						active: selectedKey === keyValue,
+						focus: activeDescendant === keyValue && isFocus,
+						hover: activeDescendant === keyValue && !isFocus,
+					})}
+					disabled={disabled}
+					id={String(keyValue)}
+					onClick={() => onSelectionChange(keyValue!)}
+					role="option"
+					tabIndex={-1}
+				>
+					{selectedKey === keyValue && (
+						<span className="dropdown-item-indicator-start">
+							<Icon symbol="check-small" />
+						</span>
+					)}
+
+					{children}
+				</button>
+			)}
```

Comment:
I think we can simplify this here with the dynamic component to avoid code repetition. For example:

```jsx
const As = href ? 'a' : 'button';

return (
    <As
    	{...otherProps}
    	{...hoverProps}
    	aria-describedby={ariaDescribedby}
    	aria-label={ariaLabel}
    	aria-labelledby={ariaLabelledby}
    	aria-posinset={ariaPosInset}
    	aria-selected={selectedKey === keyValue}
    	aria-setsize={ariaSetSize}
    	className={classNames('dropdown-item', {
    		active: selectedKey === keyValue,
    		focus: activeDescendant === keyValue && isFocus,
    		hover: activeDescendant === keyValue && !isFocus,
    	})}
        href={href.toString()}
    	disabled={disabled}
    	id={String(keyValue)}
    	onClick={() => onSelectionChange(keyValue!)}
    	role="option"
    	tabIndex={-1}
    >
    	{selectedKey === keyValue && (
    		<span className="dropdown-item-indicator-start">
    			<Icon symbol="check-small" />
    		</span>
    	)}
    
    	{children}
    </As>   
);
```

You just have to check if adding the `href` can cause any type problems, if so you can try, `...(href ? {href} : {})`.

Reply:
Thanks!!